FOCUS #262: Update glossary (#268)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/glossary.md

# Glossary

## <a name="glossary:adjustment">Adjustment</a>

A charge representing a modification to billing data to account for certain events or circumstances not previously captured, or captured incorrectly. Examples include: billing errors, service disruptions, or pricing changes.

## <a name="glossary:amortization">Amortization</a>

The distribution of upfront costs over time to accurately reflect the consumption or benefit derived from the associated resources or services. Amortization is valuable when the commitment period (time duration of the cost) extends beyond the granularity of the source report.

> ## <a name="glossary:availability-zone">Availability Zone</a>

Reviewer: I was trying to avoid having the glossary entries show up with link formatting. Let me see if I can mess with the CSS but this doesn't look good IMO. It looks like it's a link to something else when we're trying to just drop an anchor to this term. That's why I tried to just put the link around the * (empty a-tag) doesn't work on PDF, it works on the HTML rendering though.

*This section is non-normative.*

## Adjustment

A charge representing a modification to billing data to account for certain events or circumstances not previously captured, or captured incorrectly. Examples include: billing errors, service disruptions, or pricing changes.

## Amortization

The gradual recognition of upfront costs over the duration of time that the good or service is consumed. An upfront payment made toward a commitment based discount can be amortized over commitment life. Amortization is valuable when the commitment period (time duration of the cost) extends beyond the granularity of the source report.

> ## Billed Cost

Reviewer 2: @mike-finopsorg Is this going to conflict with the anchors that are generated for the columns? (Same for any other definitions that have the same names as these headers?)

Reviewer: Here's what we need to add: 
<a class="glossary" name="glossary:billedcost">&nbsp;</a>

Im not sure if the class glossary will do anything. We may need to override that in our main css if needed. Based on a quick look, I'm not sure where the class for the standard links from the table of contents (class="toc-anchor") is coming from. For now, go with this please.

> Geograhically separated locations containing a data center or cluster of data centers. Each availability zone (AZ) should have its own power, cooling, and networking, to provide redundancy and fault tolerance.

Reviewer 3: ```suggestion
A collection of geographically-separated locations containing a data center or cluster of data centers. Each availability zone (AZ) should have its own power, cooling, and networking, to provide redundancy and fault tolerance.
```

## Billing Account

> A mandatory account established with a service provider to manage and pay for all charges associated with an invoice. A billing account might have sub-accounts, all of whose costs are consolidated to the billing account.

Reviewer: From BillingAccountId/Name: 
"A billing account is a container for resources and/or services that are billed together in an invoice."

The initial thinking was that we'd use the definition from the spec (don't have to if we believe this is better). I definitely like the second sentence. Should it say 
"all of whose costs are consolidated to and invoiced at the billing account."

## Billing Currency

> The specific currency in which invoices or bills for services are issued and payments are expected to be made.

Reviewer: From the spec:
```suggestion
Billing Currency is an identifier that represents the currency that a charge for resources and/or services was billed in.
```

## Billing Period

> The time interval over which a customer's usage of services is measured for billing purposes. For most cloud providers, the billing period is a month.

Reviewer: From the spec:
```suggestion
A Billing period represents the time window for which an organization has or will receive an invoice for. The time window is inclusive of the start date and exclusive of the end date.
```

Author: @udam-f2 Is that second sentence always guaranteed to be true? Or is this definition specific to our implementation of `Billing Period`?

## Block Pricing

> Pricing model where the cost of a particular resource or service is determined based on predefined blocks or tiers of usage. In these scenarios, the Pricing Quantity can be different from the Usage Quantity.

Reviewer 2: Let's not use "pricing model" since block pricing is a bit different than how we discuss pricing models.
Block pricing is associated with SKUs, not resources or services.
nit: Replacing "blocks" with "chunks" to use a different word. If you prefer blocks, that's also fine.
```suggestion
Pricing approach where the cost of a particular SKU is determined based on predefined chunks of units. When a SKU uses block pricing, the Pricing Quantity can be different from the Usage Quantity.
```

> Pricing approach where the cost of a particular resource or service is determined based on predefined quantities or tiers of usage. In these scenarios, the Pricing Quantity can be different from the Usage Quantity.

Reviewer: ```suggestion
Pricing approach where the cost of a particular resource or service is determined based on predefined quantities or tiers of usage. In these scenarios, the Pricing Unit and the corresponding Pricing Quantity can be different from the Usage Unit and Usage Quantity.
```

## Charge

> A line item in a FOCUS compatible cost and usage data associated with the usage of a service.

Reviewer 2: ```suggestion
A row in a FOCUS compatible cost and usage data associated with the usage of a service.
```

> A row in a FOCUS compatible cost and usage data associated with the usage of a service.

Reviewer: ```suggestion
A row in a FOCUS compatible cost and usage dataset.
```

## Commitment

> This refers to a customer's agreement to purchase or use a specific quantity or level of a service over a defined period. The customer commits to a minimum usage or spending threshold.

Reviewer 2: Generic "commitment" references will also include capacity commitments.

## Commitment-Based Discount

> Also known as Commitment Discount, this is a pricing strategy or incentive offered by a service provider to encourage customers to commit to a certain level of usage or purchase over a specified period. Offerings vary in specificity of commitment, penalties for underutilization, and possible remedies.

Reviewer: ```suggestion
Also known as Commitment Discount, this is a pricing strategy or incentive offered by a service provider to encourage customers to commit to a certain level of usage or spend, over a specified period. Offerings vary in specificity of commitment, penalties for underutilization, and possible remedies.
```

## CSP

Cloud Service Provider (e.g. Amazon Web Services, Microsoft Azure, Google Cloud Platform).

## Dimension

Specification-defined categorical attribute that provides context or categorization to billing data.

## Effective Cost

> A comprehensive and realistic measure of the true cost of a service, inclusive of all reduced rates, discounts, and impacts of amortization.

Reviewer 3: ```suggestion
A measure of the true cost of a service, inclusive of all reduced rates, discounts, and impacts of amortization.
```

Feels like hyperbole with "comprehensive and realistic". Defer to others.

> A comprehensive and realistic measure of the true cost of a service, inclusive of all reduced rates, discounts, and impacts of amortization.

Reviewer: From the spec:
```suggestion
The cost inclusive of the impacts of all reduced rates and discounts, augmented with the amortization of relevant purchases (one-time or recurring) paid to cover future eligible charges. The amortized portion included should be proportional to the [PricingQuantity](#pricingquantity) and the time granularity of the data. This price is denominated in the [Billing Currency](#glossary:billingcurrency).
```

Reviewer 3: Added aspects of this comment

## Finalized Tag

> A tag with one tag value chosen from a set of possible tag values after being processed by a set of provider-defined or user-defined rules.

Reviewer 4: We need to decide whether and how we'll be linking to column definitions.  I would love to see something like "for more, see the [Finalized Tags](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/blob/a18d86c50299c02a6237b23203b7481dd474af7b/specification/dimensions/tags.md#finalized-tags) section of the [Tags](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/blob/a18d86c50299c02a6237b23203b7481dd474af7b/specification/dimensions/tags.md) column definition.  That would be helpful context for me to know more about this term, which yields nothing upon Googling.

Reviewer: @cnharris10 was the plan to move the content we put in the column spec here and link to here?

Reviewer 3: We've gone back and forth on this.  We've settled on disclosing terms in the spec and linking to the glossary.  I like Shawn's suggestion but need buy-in.

Reviewer 3: I'm going to leave as is and adjust potentially in the preview phase.

## FOCUS

> FinOps Cost and Usage Specification - An effort to build an open specification for cloud billing data.

Reviewer 3: Same comment as MSP, add as part of the title?

> FinOps Cost and Usage Specification - An effort to build an open specification for cloud billing data.

Reviewer: ```suggestion
FinOps Cost and Usage Specification - An open source specification that defines requirements for billing data.
```

## Interruptible

> Category of compute resources that can be paused or terminated by the CSP within certain criteria, often available for discounted pricing when compared to the equivalent non-interruptible resource.

Reviewer: ```suggestion
Category of compute resources that can be paused or terminated by the CSP within certain criteria, often advertised at reduced unit pricing when compared to the equivalent non-interruptible resource.
```

## List Unit Price

> Provider published unit price for a single pricing measurement unit.

Reviewer: From the spec:
```suggestion
The suggested provider-published unit price for a single [Pricing Unit](#pricingunit) of the associated SKU, exclusive of any discounts. This price is denominated in the [Billing Currency](#glossary:billingcurrency).
```

## Metric

> A quantifiable, standardized measure used to assess, evaluate, or track the performance, quality, or characteristics of a system, process, or service.

Reviewer: I think from our FOCUS perspective, this is referring to a quantitative measurement that can be used for aggregations?

## MSP

> Managed Service Provider - A company or organization that provides outsourced management and support of a range of IT services, such as network infrastructure, cybersecurity, cloud computing, and more.

Reviewer 3: Should we include "Managed Service Provider" as part of the term?

MSP (Managed Service Provider)

## On-Demand

A term that describes a service that is available and provided immediately or as needed, without requiring a pre-scheduled appointment or prior arrangement. In Cloud Computing, virtual machines can be created and terminated as needed, i.e. on demand.

## Practitioner

> An individual whose charter is to enhance business value from cloud infrastructure, by balancing innovation, velocity, and cost efficiency.

Reviewer: Suggestion only :) ignore if not useful
```suggestion
An individual who performs FinOps within an organization to maximize the business value of using cloud and cloud-like services.
```

> Any individual whose charter is to enhance business value from cloud infrastructure, by balancing innovation, velocity, and cost efficiency.

Reviewer 3: ```suggestion
An individual whose charter is to enhance business value from cloud infrastructure, by balancing innovation, velocity, and cost efficiency.
```

Every other definition uses "A"/"An" over "Any"

## Provider

> An entity that delivers services to consumers or clients i.e. a company that offers cloud computing, managed IT services, web hosting, or other technology-related services.

Reviewer: From the spec:
```suggestion
A Provider is an entity that made the resources and/or services available for purchase.
```

> ## Price list

Reviewer 4: Should be consistent with capitalization in glossary entries.

```suggestion
## Price List
```

> Comprehensive list of prices for offerings from a CSP, published by the CSP.

Reviewer: We should make this not cloud specific. Should this be related to SKU or SKU Price?

Reviewer 3: Added aspects of this comment

> Comprehensive list of prices for offerings from a CSP, published by the CSP.

Reviewer 3: ```suggestion
A comprehensive list of prices for offerings from a CSP, published by the CSP.
```

To normalize with other definitions starting with an article

## Resource

> An provisionable infrastructure component that is available to a customer in a cloud environement e.g. virtual machine, storage volume etc.

Reviewer: Any way to make this not just cloud/saas specific? In the context of FOCUS, this could apply to on-prem, labor, licenses etc.

Reviewer 3: Added aspects of this comment

## Row

> An entry in a billing data set representing the consumption of a resource or other measurable activity.

Reviewer: ```suggestion
A row in a FOCUS compatible cost and usage dataset.
```

## Service

> A capability defined and offered by cloud service providers to satisfy certain needs, often through activation of resources e.g. the capability of creating a virtual machine or a storage volume are services.

Reviewer: From the spec:
```suggestion
A service represents an offering that can be purchased from a provider (e.g., cloud virtual machine, SaaS database, professional services from a systems integrator). A service offering can include various types of usage or other charges. For example, a cloud database service may include compute, storage, and networking charges.
```

> ## SKU

Reviewer 3: ```suggestion
## Sku
```

I think this is the form we use the specs?

Reviewer 4: No... Display names will be `SKU`, column IDs will be `Sku`.  [Here](https://finopsfoundation.slack.com/archives/C053CNCRVCL/p1699049826828449) is a link to the poll and resolution.

> Construct composed of the common properties of a product offering associated with one or many SKU Prices.

Reviewer: Should we mention: "this can be referenced on a catalog or price list published by a provider to look up detailed information about the SKU. The composition of the properties associated with the SKU ID may differ across providers. Some providers may not support the SKU construct and instead associate all such properties directly with the SKU Price."

Reviewer 3: Added aspects of this comment

## SKU Price

> Unit price used to calculate a charge. Any SKU Price can only be associated to one SKU.

Reviewer: Should we mention that "this can be referenced on a price list published by a provider to look up detailed information, including a corresponding list unit price. The composition of the properties associated with the SKU Price may differ across providers. "

Reviewer 3: Added aspects of this comment

## Sub Account

> A child account associated with a billing account, for further organization and separation of resources and services.

Reviewer: From spec:
```suggestion
A sub account is an optional provider-supported construct for organizing resources and/or services connected to a billing account. Sub accounts are commonly used for scenarios like grouping based on organizational constructs, access management needs and cost allocation strategies. Sub accounts must be associated with a billing account as they do not receive invoices.
```

## Tag

A metadata label assigned to a resource to provide information about it or to categorize it for organizational and management purposes.

## Tag Source

> A Resource or Provider-defined construct for grouping resources and/or other Provider-defined constructs.

Reviewer: ```suggestion
A Resource or Provider-defined construct for grouping resources and/or other Provider-defined construct that a Tag can be assigned to.
```

> ## Zone

Reviewer 2: Shouldn't this be "Availability zone"?
